src/sync/channel.zig

```diff
@@ -5,14 +5,29 @@ const Backoff = @import("backoff.zig").Backoff;
 const Atomic = std.atomic.Value;
 const Allocator = std.mem.Allocator;
 
+const ExitCondition = sig.sync.ExitCondition;
+
 pub fn Channel(T: type) type {
     return struct {
         head: Position,
         tail: Position,
         closed: Atomic(bool) = Atomic(bool).init(false),
         allocator: Allocator,
-        mutex: std.Thread.Mutex = .{},
-        condition: std.Thread.Condition = .{},
+        event: std.Thread.ResetEvent = .{},
+        send_hook: ?*SendHook = null,
+
+        pub const SendHook = struct {
+            /// Called before a value is sent to a channel.
+            /// If it returns false, the value ia not pushed to the channel.
```

Comment:
```suggestion
            /// If it returns false, the value is not pushed to the channel.
```

Reply:
https://github.com/Syndica/sig/pull/486/commits/767607c3c46beb88cf4533d9bbabba71b9b3e634 (removed `beforeSend` for now, reasoning: https://github.com/Syndica/sig/pull/486#discussion_r1917161745)